operations/billingautomations.md

```diff
@@ -0,0 +1,591 @@
+      "Assignments": null
+    }
+  ]
+}
+```
+
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `ClientToken` | string | required | Token identifying the client application. |
+| `AccessToken` | string | required | Access token of the client application. |
+| `Client` | string | required | Name and version of the client application. |
+| `EnterpriseId` | string | optional | Unique identifier of the enterprise. Required when using [Portfolio Access Tokens](../concepts/multi-property.md), ignored otherwise. |
+| `BillingAutomationId` | string | required | Unique identifier of the billing automation. |
+| `AssignmentsToAdd` | array of [Billing automation assignment add parameters](billingautomations.md#billing-automation-assignment-add-parameters) | optional, max 20 items | List of parameters adding new assignments (or null should it not be updated). |
+| `AssignmentsToUpdate` | array of [Billing automation assignment update parameters](billingautomations.md#billing-automation-assignment-update-parameters) | optional, max 20 items | List of parameters updating existing assignments (or null should it not be updated). |
+| `AssignmentIdsToRemove` | array of string | optional, max 100 items | List of unique identifiers of assignments, which should be removed (or null should it not be updated). |
+
+#### Billing automation assignment update parameters
+
+| Property | Type | Contract | Description |
+| :-- | :-- | :-- | :-- |
+| `BillingAutomationAssignmentId` | string | required | Unique identifier of the billing automation assignment. |
+| `RoutedItemTypes` | [Billing automation item types parameters options](billingautomations.md#billing-automation-item-types-parameters-options) | optional | Type of items that are going to be routed (or `null` should it not be updated). |
+| `RateIds` | [GuidIEnumerableUpdateValue](_object.md#array-of-strings-update-value) | optional, max length 100 characters | Unique identifiers of `Rates` that will be applied as billing automation condition (or `null` should it not be updated). |
+| `ProductCategoryIds` | [GuidIEnumerableUpdateValue](_object.md#array-of-strings-update-value) | optional, max length 100 characters | Unique identifiers of `Product Categories` that will be applied as billing automation condition (or `null` should it not be updated). |
+| `ProductIds` | [GuidIEnumerableUpdateValue](_object.md#array-of-strings-update-value) | optional, max length 100 characters | Unique identifiers of `Products` that will be applied as billing automation condition (or `null` should it not be updated). |
+| `AccountingCategories` | [BillingAutomationAccountingCategoryAssignmentIEnumerableUpdateValue](billingautomations.md#billingautomationaccountingcategoryassignmentienumerableupdatevalue) | optional, max length 100 characters | List of `Accounting Categories` with their `ItemTypes` that will be applied as billing automation condition (or `null` should it not be updated). |
```

Comment:
Hey, we can't have ridiculous long names like `BillingAutomationAccountingCategoryAssignmentIEnumerableUpdateValue`, that is going to make a mess of the documentation, can't we come up with some more sensible alternatives?

Reply:
@MikeAdamsMews I added summary, operations readme and changelog entries.

Regarding the long type names - I made changes to OAS so that the titles for various "update value" types derive from the titles of their nested types. They are still crazy long, but with spaces and should wrap into multiple lines when displayed in the documentation page. Is this good enough?

Reply:
Thank you @jonas-paul, I think they will render a lot better with spaces.